x/protocolpool/depinject.go

```diff
@@ -53,7 +53,12 @@ func ProvideModule(in ModuleInputs) ModuleOutputs {
 		authority = authtypes.NewModuleAddressOrBech32Address(in.Config.Authority)
 	}
 
-	k := keeper.NewKeeper(in.Codec, in.Environment, in.AccountKeeper, in.BankKeeper, in.StakingKeeper, authority.String())
+	authorityAddr, err := in.AccountKeeper.AddressCodec().BytesToString(authority)
+	if err != nil {
+		panic(err) // TODO
```

Comment:
I don't like this at all :/

Reply:
since this is only called at startup its fine. Not the best agree, but we would panic up the stack anyways